advanced/wallets/react-wallet-v2/src/pages/settings.tsx

```diff
@@ -55,30 +56,51 @@ export default function SettingsPage() {
 
       <Divider y={2} />
 
-      <Text h4 css={{ marginBottom: '$5' }}>
-        Smart Account Sponsorship (Pimlico)
-      </Text>
-      <Row justify="space-between" align="center">
-        <Switch
-          checked={smartAccountSponsorshipEnabled}
-          onChange={SettingsStore.toggleSmartAccountSponsorship}
-          data-testid="settings-toggle-smart-account-sponsorship"
-        />
-        <Text>{smartAccountSponsorshipEnabled ? 'Enabled' : 'Disabled'}</Text>
-      </Row>
-
-      <Divider y={2} />
-
-      <Text h4 css={{ marginBottom: '$5' }}>
-        Smart Account Enabled
-      </Text>
-      <Row justify="space-between" align="center">
-        <Switch
-          checked={smartAccountEnabled}
-          onChange={SettingsStore.toggleSmartAccountEnabled}
-          data-testid="settings-toggle-smart-account-enabled"
-        />
-        <Text>{smartAccountEnabled ? 'Enabled' : 'Disabled'}</Text>
+      <Row>
+        <Col>
+          <Text h4 css={{ marginBottom: '$5' }}>
+            Smart Accounts
+          </Text>
+          <Text h4 css={{ marginBottom: '$5' }}>
+            Enabled
+          </Text>
+          <Row justify="space-between" align="center">
+            <Switch
+              checked={smartAccountEnabled}
+              onChange={SettingsStore.toggleSmartAccountEnabled}
+              data-testid="settings-toggle-smart-account-enabled"
+            />
+            <Text>{smartAccountEnabled ? 'Enabled' : 'Disabled'}</Text>
+          </Row>
+
+          {smartAccountEnabled ? (
+            <>
+              <Text h4 css={{ marginBottom: '$5', marginTop: '$5' }}>
+                ZeroDev Smart Account
+              </Text>
+              <Row justify="space-between" align="center">
+                <Switch
+                  checked={kernelSmartAccountEnabled}
+                  onChange={SettingsStore.toggleKernelSmartAccountsEnabled}
+                  data-testid="settings-toggle-smart-account-sponsorship"
+                />
+                <Text>{kernelSmartAccountEnabled ? 'Enabled' : 'Disabled'}</Text>
+              </Row>
+
+              <Text h4 css={{ marginBottom: '$5', marginTop: '$5' }}>
+                Sponsorship (Pimlico)
```

Comment:
Should be (`Kernel`) right?

Reply:
That currently does nothing as we don't use previous Lib, working on removing it completely and implementing Safe from scratch (with pimlico probably)